pallets/claim/src/lib.rs

```diff
@@ -164,89 +180,172 @@ decl_error! {
 		/// The module doesn't set relyer
 		NoRelayer,
 	}
-}
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		// Errors must be initialized if they are used by the pallet.
-		type Error = Error<T>;
-		fn deposit_event() = default;
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
 
-		#[weight = 0]
-		pub fn change_relayer(origin, new_relayer: T::AccountId) -> dispatch::DispatchResult {
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(0 + T::DbWeight::get().reads_writes(1,1))]
+		pub fn change_relayer(
+			origin: OriginFor<T>,
+			new_relayer: T::AccountId,
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 			let old_relayer = Relayer::<T>::get();
-			ensure!(Some(&new_relayer) != old_relayer.as_ref(), Error::<T>::RelayerNotChanged);
-			Relayer::<T>::put(&new_relayer);
-			Self::deposit_event(RawEvent::RelayerChanged(new_relayer));
-			Ok(())
+			if old_relayer.is_some() {
+				ensure!(
+					Some(&new_relayer) != old_relayer.unwrap().as_ref(),
+					Error::<T>::RelayerNotChanged
+				);
+			}
+
+			Relayer::<T>::put(Some(&new_relayer));
+			Self::deposit_event(Event::RelayerChanged(new_relayer));
+			Ok(().into())
 		}
 
-		#[weight = 0]
-		pub fn store_erc20_burned_transactions(origin, height: u64, claims:Vec<(EthereumTxHash, EthereumAddress, BalanceOf<T>)>) -> dispatch::DispatchResult {
+		#[pallet::weight(0 + T::DbWeight::get().reads_writes(1,1))]
+		pub fn store_erc20_burned_transactions(
+			origin: OriginFor<T>,
+			height: u64,
+			claims: Vec<(EthereumTxHash, EthereumAddress, BalanceOf<T>)>,
+		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 			let relayer = Relayer::<T>::get();
 			ensure!(relayer.is_some(), Error::<T>::NoRelayer);
-			ensure!(Some(&who) == relayer.as_ref(), Error::<T>::CallerNotRelayer);
+			ensure!(
+				Some(&who) == relayer.unwrap().as_ref(),
+				Error::<T>::CallerNotRelayer
+			);
```

Comment:
Same as before. Don't `unwrap()` without a good reason because it may crash the runtime (and the runtime is supposed to never crash)